### PR TITLE
fix: right-click paste for images in clipboard (Issue #8826)

### LINKED ISSUE #8826: Right click paste command does not work for images
Tried to paste with right click and paste command, but it doesn't work.

### PR DESCRIPTION
**Pull request by shreyjay0**
This PR fixes the issue with pasting images from the clipboard using the right-click context menu. Previously, users were unable to paste copied images from the clipboard when using the right-click action, which hindered the usability of the application. While pasting with `Ctrl + V` was functioning correctly, the right-click custom paste option was not working as intended. This fix addresses that issue and improves the overall user experience.

### Related Issue
Fixes #8826.

### CODE REVIEW

File: packages/excalidraw/clipboard.ts [hunk 1]
```diff
@@ -270,6 +255,34 @@ export const readSystemClipboard = async () => {
...
+  try {
+    if (navigator.clipboard?.read) {
+      const readClipboard = await navigator.clipboard?.read();
+      const textItem = readClipboard.find((item) => 
+        item.types.includes("text/plain")
+      );
+      if (textItem) {
+        const readText = await navigator.clipboard.readText();
+        if (readText) {
+          return { "text/plain": readText };
```
Comment (Mrazator) [inline packages/excalidraw/clipboard.ts:267]: This seems a bit strange, as it assumes all clipboard items are of the same exact type. Could we extend it to support pasting any types together? At the same time, when one type fails, the whole paste should not fail and continue.

File: packages/excalidraw/clipboard.ts [hunk 2]
```diff
@@ -319,9 +332,21 @@ const parseClipboardEvent = async (
       return mixedContent;
     }
 
+    const imageType = event.clipboardData?.types[0]?.startsWith("image/") ? event.clipboardData?.types[0] : "image/png";
```
Comment (Mrazator) [inline packages/excalidraw/clipboard.ts:335]: Similar to the above comments. Should we assume all clipboard items are of the same type? Should we default to png?

File: packages/excalidraw/clipboard.ts [hunk 3]
```diff
@@ -270,6 +255,34 @@ export const readSystemClipboard = async () => {
     throw error;
   }
 
+  try {
+    if (navigator.clipboard?.read) {
```
[the remark this answers is not in the document]
Comment (shreyjay0) [reply packages/excalidraw/clipboard.ts:259]: Apologies, I missed a section of the code that handles this logic, as I mistakenly thought it was part of the error handling, which only considers a single paste type. I'll push a follow-up change to address this.

File: packages/excalidraw/clipboard.ts [hunk 4]
```diff
@@ -270,6 +255,34 @@ export const readSystemClipboard = async () => {
...
+        const readText = await navigator.clipboard.readText();
+        if (readText) {
+          return { "text/plain": readText };
+        }
+      }
+      const imageItem = readClipboard.find((item) => 
+        item.types.some(type => type.startsWith("image/"))
+      );
+      if (imageItem) {
+        const imageType = imageItem.types.find(type => type.startsWith("image/")) || "image/png";
```
[the remark this answers is not in the document]
Comment (shreyjay0) [reply packages/excalidraw/clipboard.ts:274]: The image types are defined using MIME types. I'll update it to refer to 'type' directly.

File: packages/excalidraw/clipboard.ts (hunk 1 above)
Comment (dwelle) [reply packages/excalidraw/clipboard.ts]: yeah, I think this is some legacy code. I've now put it as a fallback to `read()` instead.

**Comment by dwelle** [comment]
Thanks @shreyjay0! I've made a tweak to remove the image handling logic from `parseClipboard` as that function is pretty much meant just for text (your code there wasn't doing anything AFAIK as the native `paste` event already works with images using a different code path).

I've also done some refactor.

We're good to go!

**Review by dwelle — APPROVED** [review]

**Pull request merged**